addons/binding/org.openhab.binding.samsungtv/src/main/java/org/openhab/binding/samsungtv/internal/protocol/RemoteControllerWebSocket.java

```diff
@@ -0,0 +1,678 @@
+
+            } catch (Exception e) {
+                logger.warn("Cannot connect to websocket remote control interface: " + e.getMessage());
+                throw new RemoteControllerException(e);
+            }
+        }
+        connectWebSockets();
+    }
+
+    private void connectWebSockets() {
+        try {
+            webSocketRemote.connect(new URI("ws", null, host, port, WS_ENDPOINT_REMOTE_CONTROL, "name=openhab", null));
+        } catch (RemoteControllerException | URISyntaxException e) {
+            logger.warn("Problem connecting to remote websocket", e);
+        }
+        try {
+            webSocketArt.connect(new URI("ws", null, host, port, WS_ENDPOINT_ART, "name=openhab", null));
+        } catch (RemoteControllerException | URISyntaxException e) {
+            logger.warn("Problem connecting to artmode websocket", e);
+        }
+        try {
+            webSocketV2.connect(new URI("ws", null, host, port, WS_ENDPOINT_V2, "name=openhab", null));
+        } catch (RemoteControllerException | URISyntaxException e) {
+            logger.warn("Problem connecting to V2 websocket", e);
+        }
+    }
+
+    private void closeConnection() throws RemoteControllerException {
+        logger.debug("RemoteControllerWebSocket closeConnection");
+
+        try {
+            webSocketRemote.close();
+            webSocketArt.close();
+            webSocketV2.close();
+            client.stop();
+        } catch (Exception e) {
+            throw new RemoteControllerException(e);
+        }
+    }
+
+    @Override
+    public void close() throws Exception {
+        logger.debug("RemoteControllerWebSocket close");
+        closeConnection();
+    }
+
+    /**
+     * Websocket base class
+     *
+     * @author Arjan Mels
+     *
+     */
+    class WebSocketBase extends WebSocketAdapter {
+        boolean isConnecting = false;
+
+        @Override
+        public void onWebSocketClose(int statusCode, String reason) {
+            logger.debug("{} connection closed: {} - {}", this.getClass().getSimpleName(), statusCode, reason);
+            super.onWebSocketClose(statusCode, reason);
+            isConnecting = false;
+        }
+
+        @Override
+        public void onWebSocketError(Throwable error) {
+            logger.error("{} connection error: {}", this.getClass().getSimpleName(), error.getMessage());
+            super.onWebSocketError(error);
+            isConnecting = false;
+        }
+
+        void connect(URI uri) throws RemoteControllerException {
+            if (isConnecting || isConnected()) {
+                logger.trace("{} already connecting or connected", this.getClass().getSimpleName());
+                return;
+            }
+
+            try {
+                logger.debug("{} connecting to: {}", this.getClass().getSimpleName(), uri);
+
+                isConnecting = true;
+                client.connect(this, uri, new ClientUpgradeRequest());
+            } catch (Exception e) {

```

Comment:
Do you really need to catch the Exception-class or are there more specific exceptions you could catch like the IOException or maybe only RuntimeException's (which is not that good either). The current code and also catching RuntimeException's might 'hide' programming errors like NullPointerExceptions occurring.

https://www.google.nl/search?q=java%20why%20not%20to%20catch%20exception

Reply:
Changed it to IOException and only included the connect line.